pingpong/server.py

```
@@ -13240,6 +13246,159 @@ async def update_external_login_provider(
+    secret_provided = bool(req.client_secret)
+    enabling_connector = not config_obj.enabled and req.enabled
+    new_secret = req.client_secret if secret_provided else config_obj.client_secret
+
+    if host_changed or client_id_changed or secret_provided or enabling_connector:
+        try:
+            connector = get_connector(config_obj.service)
+        except ConnectorNotRegistered:
+            raise HTTPException(
+                status_code=400,
+                detail=f"Unknown connector service '{config_obj.service}'.",
+            )
```

Ruff (0.15.13)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)